### PR TITLE
Add authorization support for the API

### PR DESCRIPTION
**Pull request by viktor-yakubiv**
1.  Adds `API_KEY` project environment variable that reflects `CORE_API_KEY` system environment variable.

2.  Makes the `API_KEY` required in the environment configuration.

3.  Adds `Authorization` header as default for all API requests.

### CODE REVIEW

**Review by mcancellieri — APPROVED** [review]
Looks good to me. Thanks :)

File: env.config.js [hunk 1]
```diff
@@ -13,14 +13,15 @@ const production = {
...
-  ['API_URL'].forEach((param) => {
+  ['API_URL', 'API_KEY'].forEach((param) => {
     if (config[param] == null) throw new Error(`${param} is not configured.`)
   })
 
 const env = { local, development, production }
 const config = {
   ...env.production,
   ...env[NODE_ENV],
+  API_KEY: process.env.CORE_API_KEY,
```
Comment (viktor-yakubiv) [inline env.config.js:24]: @mcancellieri I am not sure about this mapping. I prefer API_KEY in the code for consistency. Also, I assume, we use only one API, so there is no need to clarify it with prefix but who knows...

I tend to choose either one or another.

Comment (mcancellieri) [reply env.config.js:24]: ok, my vote goes for API_KEY too...
The CORE part is redundant, if you will need other API Keys you can prefix them.

Comment (viktor-yakubiv) [reply env.config.js:24]: Thank you for your comment. I will change them on Monday 🙂

**Pull request merged**